client/collector/src/main/java/org/hawkular/apm/client/collector/DefaultTraceCollector.java

```diff
@@ -117,56 +131,77 @@ public void setConfigurationService(ConfigurationService cs) {
-                                BusinessTxnConfig btc = stringBusinessTxnConfigEntry.getValue();
-
-                                if (btc.isDeleted()) {
-                                    if (log.isLoggable(Level.FINER)) {
-                                        log.finer("Removing config for btxn '" + stringBusinessTxnConfigEntry.getKey() + "' = " + btc);
-                                    }
-
-                                    filterManager.remove(stringBusinessTxnConfigEntry.getKey());
-                                    processorManager.remove(stringBusinessTxnConfigEntry.getKey());
-                                } else {
-                                    if (log.isLoggable(Level.FINER)) {
-                                        log.finer("Changed config for btxn '" + stringBusinessTxnConfigEntry.getKey() + "' = " + btc);
-                                    }
-
-                                    filterManager.init(stringBusinessTxnConfigEntry.getKey(), btc);
-                                    processorManager.init(stringBusinessTxnConfigEntry.getKey(), btc);
+        if (log.isLoggable(Level.FINER)) {
+            log.finer("Configuration refresh cycle (in seconds) = " + refresh);
+        }
+
+        if (refresh != null) {
+            Executors.newSingleThreadScheduledExecutor(new ThreadFactory() {
+                public Thread newThread(Runnable r) {
+                    Thread t = Executors.defaultThreadFactory().newThread(r);
+                    t.setDaemon(true);
+                    return t;
+                }
+            }).scheduleAtFixedRate(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        Map<String, BusinessTxnConfig> changed = configurationService.getBusinessTransactions(null,
+                                configLastUpdated);
+
+                        for (Map.Entry<String, BusinessTxnConfig> stringBusinessTxnConfigEntry : changed.entrySet()) {
+                            BusinessTxnConfig btc = stringBusinessTxnConfigEntry.getValue();
+
+                            if (btc.isDeleted()) {
+                                if (log.isLoggable(Level.FINER)) {
+                                    log.finer("Removing config for btxn '" + stringBusinessTxnConfigEntry.getKey() + "' = " + btc);
                                 }
 
-                                if (btc.getLastUpdated() > configLastUpdated) {
-                                    configLastUpdated = btc.getLastUpdated();
+                                filterManager.remove(stringBusinessTxnConfigEntry.getKey());
+                                processorManager.remove(stringBusinessTxnConfigEntry.getKey());
+                            } else {
+                                if (log.isLoggable(Level.FINER)) {
+                                    log.finer("Changed config for btxn '" + stringBusinessTxnConfigEntry.getKey() + "' = " + btc);
```

Comment:
This log could be there only once, `btc.toString()` contains deleted attribute. On the other hand explicit logging might be better for redibility of the log file.